packages/mobile/src/components/button/button.tsx

```diff
@@ -1,17 +1,80 @@
 import React, { FC } from 'react';
-import { Pressable } from 'react-native';
+import { Pressable, View } from 'react-native';
+import { ButtonType } from '~/common/enums/enums';
 import { Text } from '~/components/components';
 import { styles } from './styles';
 
+//mock before update global theme color
+enum AppColor {
+  YELLOW_100 = '#F1B313',
+  YELLOW_200 = '#FFB800',
+  GRAY_200 = '#DFDFDF',
+  WHITE = '#FFFFFF',
+}
+////
+
 type Props = {
   label: string;
+  type?: ButtonType;
+  isDisabled?: boolean;
+  isOutlined?: boolean;
   onPress: () => void;
 };
 
-const Button: FC<Props> = ({ label, onPress }) => {
+const Button: FC<Props> = ({
+  label,
+  type = ButtonType.MAIN,
+  isDisabled,
+  isOutlined,
+  onPress,
+}) => {
   return (
-    <Pressable style={styles.button} onPress={onPress}>
-      <Text>{label}</Text>
+    <Pressable disabled={isDisabled} onPress={onPress}>
+      <View
+        style={[
+          styles.button,
+          styles[type],
+          isOutlined
+            ? {
+                backgroundColor: AppColor.WHITE,
+                borderColor: AppColor.YELLOW_100,
+                borderWidth: 2,
```

Comment:
You may create separate styles for this options and refer to these here. Just for the readability's sake

Reply:
updated